metrics/web/docs/features/deep_links/01_deep_linking.md

```diff
@@ -399,3 +412,214 @@ class MetricsRouteInformationParser {
+
+![PageParametersModel class diagram](http://www.plantuml.com/plantuml/proxy?cache=no&fmt=svg&src=https://raw.githubusercontent.com/Flank/flank-dashboard/deep_links_design_improvements/metrics/web/docs/features/deep_links/diagrams/page_parameters_model_class_diagram.puml)
+
+##### PageParametersFactory
+Once we've implemented the required `PageParametersModel`, we should create a `PageParametersFactory` that encapsulates the creation of specific `PageParametersModel` from the given `RouteConfiguration`.
+
+Consider the following code snippet that demonstrates the `PageParametersFactory.create` method:
+```dart
+/// Returns the [PageParametersModel] based on the given [configuration].
+PageParametersModel create(RouteConfiguration configuration) {
+  final parameters = configuration.parameters;
+  
+  switch(configuration.name) {
+    case RouteName.dashboard:
+      return DashboardPageParameters.fromMap(parameters);
+    case RouteName.projectGroups:
+      /// Other page parameters handling
+  }
+}
+```
+
+The `PageParametersFactory` should be injected into `NavigationNotifier`, since the `NavigationNotifier` holds all the navigation-related logic of the application and controls the deep linking of the application. Consider the [`NavigationNotifier`](#navigationnotifier) section describing this part of the application.
+
+##### MetricsPage and MetricsPageFactory
+To be able to restore the `PageParametersModel` when the application, for example, pops a page, we should update the `MetricsPage` and `MetricsPageFactory` classes.
+
+A `MetricsPage` should store a `RouteName` used to create this page and the `PageParametersModel` as an `arguments` field. The `MetricsPage` class should also provide an overridden `copyWith` method to copy a page with the updated parameters.
+
+The `MetricsPageFactory` should be updated as well to create `MetricsPage`s with the given `RouteName` and `PageParametersModel`.
+
+The code snippet below demonstrates the `MetricsPageFactory.create` changes:
+```dart
+  /// Creates the [MetricsPage] from the given [RouteConfiguration].
+  MetricsPage create({
+    RouteName routeName, 
+    PageParametersModel pageParameters,
+  }) {
+    switch (routeName) {
+      case RouteName.loading:
+        return MetricsPage(
+          child: const LoadingPage(),
+          routeName: routeName,
+          arguments: pageParameters,
+        );
+    /// Other MetricsPages creation...
+```
+
+##### NavigationNotifier
+The `NavigationNotifier` is a class that holds the navigation logic of the Metrics Web application. The following subsections describe the main aspects needed to be able to implement the `deep linking` feature.
+
+###### Updating page parameters
+When the `NavigationNotifier` gets a new `RouteConfiguration`, it should create a new `PageParametersModel` from the received `RouteConfiguration` and notify any listeners about the `PageParametersModel` updates.
```

Comment:
:label:
Maybe "consumes" is a better word here? However, I'd like to suggest using the "receive" and replace its second appearance:
```suggestion
When the `NavigationNotifier` receives a new `RouteConfiguration`, it should create a new `PageParametersModel` using a new `RouteConfiguration` instance and notify all listeners about the updates.
```